src/net/sourceforge/kolmafia/session/GreyYouManager.java

```diff
@@ -124,6 +124,32 @@ public static GooSkill learnSkill(int skillId) {
     return skill;
   }
 
+  public static void absorbMonster(MonsterData monster, String absorbText) {
+    // This called from FightRequest after a win
+    if (monster == null || !KoLCharacter.inGreyYou()) {
+      return;
+    }
+
+    Absorption absorb = allAbsorptions.get(monster.getId());
+
+    if (absorb == null) {
+      return;
+    }
+
+    // If absorb is expected to give adventures, but neither message for adventure absorbs is found
+    if (absorb.getType() == AbsorptionType.ADVENTURES
+        && !absorbText.contains("a lot of potential energy!")
+        && !absorbText.contains("incorporate this energetic creature.")) {
+      return;
+    }
+
+    if (absorb.getType() == AbsorptionType.ADVENTURES) {
+      Preferences.increment("_greyYouAdventures", ((GooAbsorption) absorb).value);
+    }
+
+    absorbMonster(monster);
+  }
+
   public static void absorbMonster(MonsterData monster) {
```

Comment:
as this is now only called from the above method + tests, it can be package private.